pkgs/by-name/wa/waveterm/package.nix

```
@@ -0,0 +1,189 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  makeDesktopItem,
+  copyDesktopItems,
+  unzip,
+  autoPatchelfHook,
+  atk,
+  at-spi2-atk,
+  cups,
+  libdrm,
+  gtk3,
+  pango,
+  cairo,
+  libX11,
+  libXcomposite,
+  libXdamage,
+  libXext,
+  libXfixes,
+  libXrandr,
+  mesa,
+  expat,
+  libxcb,
+  alsa-lib,
+  nss,
+  nspr,
+  vips,
+  wrapGAppsHook3,
+  udev,
+  libGL,
+}:
+
+let
+  inherit (stdenv.hostPlatform) system;
+  throwSystem = throw "Unsupported system: ${system}";
+
+  pname = "waveterm";
+  version = "0.8.7";
+
+  suffix =
+    {
+      x86_64-linux = "waveterm-linux-x64-${version}.zip";
+      aarch64-linux = "waveterm-linux-arm64-${version}.zip";
+      x86_64-darwin = "Wave-darwin-universal-${version}.zip ";
+      aarch64-darwin = "Wave-darwin-arm64-${version}.zip";
+    }
+    .${system} or throwSystem;
+
+  src = fetchurl {
+    url = "https://github.com/wavetermdev/waveterm/releases/download/v${version}/${suffix}";
+    hash =
+      {
+        x86_64-linux = "sha256-pWBKZid8sumi/EP3DA5KcLnZsHsuKYK6E6NHXdWKh8s=";
+        aarch64-linux = "sha256-2paRX+OGPSEktV4S+V43ZE9UgltLYZ+Nyba5/miBQkA=";
+        x86_64-darwin = "sha256-tsqw597gQIMnQ/OPZhrWwaRliF94KyS+ryajttDLqBA=";
+        aarch64-darwin = "sha256-PD38UBSNKuv836P/py/CtrLOlddHa0+w7R20YVY4Ybc=";
+      }
+      .${system} or throwSystem;
+  };
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "waveterm";
+      exec = pname + " --no-sandbox %U";
```

```suggestion
      exec = "waveterm --no-sandbox %U";
```

executable name, unlikely to change, and using pname here is risky